src/main/java/com/rednavis/vaadin/showcase/backend/service/password/PasswordUtils.java

```diff
@@ -0,0 +1,152 @@
+package com.rednavis.vaadin.showcase.backend.service.password;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+class PasswordUtils {
```

Comment:
I hope that code was not copied out of some restricted place like previous project